app/report/models.py

```diff
@@ -50,5 +53,6 @@ class GatewayInfo(BaseModel):
 
 class StatusReport(BaseModel):
```

Comment:
Not sure what's the purpose of `Report`? seems there is no common.
also, this one should inherit from `Report` as well?

Reply:
`Report` is used for type hinting to give reports a common parent. `StatusReport` is an aggregation of other reports  + gateway info. Maybe a name change might be better for clarity?